packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -664,7 +664,12 @@ function validateClassAttr(
 
     const elmClassName = getAttribute(elm, 'class');
 
-    if (!isUndefined(className) && String(className) !== elmClassName) {
+    if (
+        !isUndefined(className) &&
+        String(className) !== elmClassName &&
+        // No mismatch if SSR className is empty and the client-side class is null
+        !(className === '' && isNull(elmClassName))
+    ) {
```

Comment:
This if is starting to get hairy, but the auto-formatting gave me the idea of adding this comment. I think this is OK.

Reply:
Could also be `(className !== '' || !isNull(elmClassName))` but I found this way more readable and easy to explain via the comment.